main/src/main/scala/sbt/Load.scala

```diff
@@ -74,8 +74,21 @@ object Load {
   def buildGlobalSettings(base: File, files: Seq[File], config: sbt.LoadBuildConfiguration): ClassLoader => Seq[Setting[_]] =
     {
       val eval = mkEval(data(config.globalPluginClasspath), base, defaultEvalOptions)
-      val imports = BuildUtil.baseImports ++ BuildUtil.importAllRoot(config.globalPluginNames)
+
       loader => {
+        val pg = PluginData(config.globalPluginClasspath, None, None)
+        val discovery = PluginDiscovery.discoverAll(pg, loader)
+
+        val detectedPlugins =
```

Comment:
Oh, hey, there is one there! I'll update the patch.

Like I said in the chat room... I don't know what I'm doing and just threw up a patch to get some validation. Please don't expect this to be me saying "here's a finished patch, please merge it right now". :)